cabal-install/Main.hs

```diff
@@ -258,6 +256,7 @@ mainWorker args = topHandler $
       , regularCmd runCommand runAction
       , regularCmd initCommand initAction
       , regularCmd configureExCommand configureAction
+      , regularCmd reconfigureCommand reconfigureAction
```

Comment:
Do we really want to make `reconfigure` a non-hidden command? IMO it'll be hard for normal users to figure out what the difference is and when to use it instead of `configure`.